metadata-ingestion/src/datahub/ingestion/source/affirm_artifact.py

```diff
@@ -0,0 +1,143 @@
+import sys
+import os
+import logging
+from dataclasses import dataclass, field
+from typing import Iterable, Sequence
+from ruamel.yaml import YAML
+
+import datahub.emitter.mce_builder as builder
+from datahub.configuration.common import ConfigModel
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.schema_classes import (
+    DatasetPropertiesClass,
+    OwnerClass,
+    OwnershipClass,
+    OwnershipTypeClass,
+)
+
+
+yaml = YAML(typ='rt')
+
+
+@dataclass
+class AffirmArtifact:
+    name: str
+    description: str
+    owner: str
+    privacy_entrypoint: str
+    retention_days: int
+    processing_purposes: Sequence[str]
+
+    def __post_init__(self):
+        if self.privacy_entrypoint is None:
+            self.privacy_entrypoint = ''
+        if self.retention_days is None:
+            self.retention_days = -1
+        if self.processing_purposes is None:
+            self.processing_purposes = []
+
+
+class AffirmArtifactSourceConfig(ConfigModel):
+    directory: str
+    filename: str
+    platform: str
+    env: str
+
+
+def iterate_artifact(directory: str) -> Iterable[AffirmArtifact]:
+    for r, _, filenames in os.walk(directory):
+        for filename in filenames:
+            if filename.endswith('.yaml'):
+                filepath = os.path.join(r, filename)
+                with open(filepath, 'r') as f:
+                    content = yaml.load(f)
+                    artifact = AffirmArtifact(
+                        name=filename.replace('.yaml', ''),
+                        description=content.get('description', ''),
```

Comment:
Can we fix any formatting issues here. Similar to https://github.com/Affirm/datahub-metadata/blob/0e4d2eec8c7687c79cff1ac50c7b4daaebe45e32/internal/code/datahub_metadata/utils/string_utils.py#L1